File: data/readable_format/cpp/file1701.cpp
#include <iostream>
#include <vector>
#include <complex>
#include <cmath> // For std::abs

// Function to check if two floating-point numbers are "close enough"
bool approxEqual(float a, float b, float epsilon = 0.001f) {
    return std::abs(a - b) < epsilon;
}

int main() {
    // Correctly setup test data with nested vectors
    std::vector<std::vector<std::vector<int>>> intVla(11, std::vector<std::vector<int>>(22, std::vector<int>(33, 1)));
    std::vector<std::vector<std::vector<float>>> realVla(11, std::vector<std::vector<float>>(22, std::vector<float>(33, 3.14f)));
    std::vector<std::vector<std::vector<std::complex<float>>>> complexVla(11, std::vector<std::vector<std::complex<float>>>(22, std::vector<std::complex<float>>(33, std::complex<float>(2.0, -3.0))));
    std::vector<std::vector<std::vector<bool>>> logicalVla(11, std::vector<std::vector<bool>>(22, std::vector<bool>(33, true)));
    std::vector<std::vector<std::vector<char>>> charactervla(11, std::vector<std::vector<char>>(22, std::vector<char>(33, 'K')));

    // Specific assignments
    intVla[4][4][4] = 42;
    realVla[4][4][4] = 4.13f;
    complexVla[4][4][4] = std::complex<float>(-3.0, 2.0);
    logicalVla[4][4][4] = false;
    charactervla[4][4][4] = 'X';

    // Assertions
    if (intVla[4][4][4] != 42) std::cerr << "Failed: intVla incorrect value." << std::endl;
    if (!approxEqual(realVla[4][4][4], 4.13f)) std::cerr << "Failed: realVla incorrect value." << std::endl;
    if (complexVla[4][4][4] != std::complex<float>(-3.0, 2.0)) std::cerr << "Failed: complexVla incorrect value." << std::endl;
    if (logicalVla[4][4][4] != false) std::cerr << "Failed: logicalVla incorrect value." << std::endl;
    if (charactervla[4][4][4] != 'X') std::cerr << "Failed: charactervla incorrect value." << std::endl;

    std::cout << "All tests passed." << std::endl;

    return 0;
}

File: data/readable_format/fortran/file1701.f90
program test_vla_primitives
  implicit none
  integer, allocatable :: intvla(:, :, :)
  real, allocatable    :: realvla(:, :, :)
  complex, allocatable :: complexvla(:, :, :)
  logical, allocatable :: logicalvla(:, :, :)
  character(len=1), allocatable :: charactervla(:, :, :)
  
  ! Allocate arrays
  allocate(intvla(11,22,33))
  allocate(realvla(11,22,33))
  allocate(complexvla(11,22,33))
  allocate(logicalvla(11,22,33))
  allocate(charactervla(11,22,33))
  
  ! Initialize arrays
  intvla = 1
  realvla = 3.14
  complexvla = cmplx(2.0, -3.0)
  logicalvla = .TRUE.
  charactervla = 'K'
  
  ! Specific assignments
  intvla(5,5,5) = 42
  realvla(5,5,5) = 4.13
  complexvla(5,5,5) = cmplx(-3.0, 2.0)
  logicalvla(5,5,5) = .FALSE.
  charactervla(5,5,5) = 'X'
  
  ! Assertions
  if (intvla(5,5,5) /= 42) stop "Failed: intvla incorrect value."
  if (realvla(5,5,5) /= 4.13) stop "Failed: realvla incorrect value."
  if (complexvla(5,5,5) /= cmplx(-3.0, 2.0)) stop "Failed: complexvla incorrect value."
  if (.not. logicalvla(5,5,5) .eqv. .FALSE.) stop "Failed: logicalvla incorrect value."
  if (charactervla(5,5,5) /= 'X') stop "Failed: charactervla incorrect value."
  
  print *, "All tests passed."
end program test_vla_primitives